ethcore/src/snapshot/service.rs

```diff
@@ -327,9 +328,74 @@ impl Service {
 
 	// replace one the client's database with our own.
 	fn replace_client_db(&self) -> Result<(), Error> {
-		let our_db = self.restoration_db();
+		let rest_db = self.restoration_db();
+
+		let cur_chain = &self.client.chain();
+		let next_db = self.restoration_db_handler.open(&rest_db)?;
+		let next_chain = BlockChain::new(Default::default(), &[], next_db.clone());
+		let next_chain_info = next_chain.chain_info();
+
+		// Try to include every block that will need to be downloaded from the current chain
+		// Break when no more blocks are available from it.
+		match (next_chain_info.ancient_block_number, next_chain_info.first_block_number) {
+			(Some(next_ancient_block), Some(next_first_block)) if next_ancient_block + 1 < next_first_block => {
```

Comment:
`ancient_block_number` is always going to be `1` (or `0`, can't remember) after a restoration, isn't it?

Reply:
the blocks we want to import from the old chain: anything available from genesis to `first_block_number`, but maintaining the invariant that only a single gap exists.

Reply:
Yep that's right, it's supposed to always be `1`. It does ensure that there is still only one gap of blocks.